app/src/main/java/com/example/sitstandtimer/data/UserPreferenceRepository.kt

```diff
@@ -17,132 +17,92 @@ class UserPreferenceRepository(
-                Log.e(TAG, "Error reading preferences.", it)
-                emit(emptyPreferences())
-            } else {
-                throw it
-            }
-        }
-        .map { preferences ->
-            preferences[LUNCH_LENGTH] ?: 60f
-        }
+        .getPreferenceWithFallback(LUNCH_LENGTH, DEFAULT_LUNCH_LENGTH)
 
     val snoozeLength: Flow<Float> = dataStore.data
-        .catch {
-            if(it is IOException) {
-                Log.e(TAG, "Error reading preferences.", it)
-                emit(emptyPreferences())
-            } else {
-                throw it
-            }
-        }
-        .map { preferences ->
-            preferences[SNOOZE_LENGTH] ?: 15f
-        }
+        .getPreferenceWithFallback(SNOOZE_LENGTH, DEFAULT_SNOOZE_LENGTH)
 
     val remoteNfcTag: Flow<String> = dataStore.data
-        .catch {
-            if(it is IOException) {
-                Log.e(TAG, "Error reading preferences.", it)
-                emit(emptyPreferences())
-            } else {
-                throw it
-            }
-        }
-        .map { preferences ->
-            preferences[REMOTE_NFC_TAG] ?: ""
-        }
+        .getPreferenceWithFallback(REMOTE_NFC_TAG, DEFAULT_NFC_TAG)
 
     val deskNfcTag: Flow<String> = dataStore.data
-        .catch {
-            if(it is IOException) {
-                Log.e(TAG, "Error reading preferences.", it)
-                emit(emptyPreferences())
-            } else {
-                throw it
-            }
-        }
-        .map { preferences ->
-            preferences[DESK_NFC_TAG] ?: ""
+        .getPreferenceWithFallback(DESK_NFC_TAG, DEFAULT_NFC_TAG)
+
+
+    suspend fun savePreferences(
+        numberOfIntervals: Int,
+        intervalLength: Float,
+        breakLength: Float,
+        lunchLength: Float,
+        snoozeLength: Float
+    ) {
+        dataStore.edit { preferences ->
+            preferences[NUMBER_OF_INTERVALS] = numberOfIntervals
+            preferences[INTERVAL_LENGTH] = intervalLength
+            preferences[BREAK_LENGTH] = breakLength
+            preferences[LUNCH_LENGTH] = lunchLength
+            preferences[SNOOZE_LENGTH] = snoozeLength
         }
+    }
 
-    suspend fun saveRemoteNfcTag(tag: String, location: String = "remote") {
-        if (location == "desk") {
-            dataStore.edit { preferences ->
-                preferences[DESK_NFC_TAG] = tag
-            }
-        } else {
-            dataStore.edit { preferences ->
-                preferences[REMOTE_NFC_TAG] = tag
-            }
+    suspend fun saveRemoteNfcTag(tag: String, location: NfcTagLocation) {
+        val key = when (location) {
+            NfcTagLocation.DESK -> DESK_NFC_TAG
+            NfcTagLocation.REMOTE -> REMOTE_NFC_TAG
+        }
+        dataStore.edit { preferences ->
+            preferences[key] = tag
         }
     }
-}
\ No newline at end of file
+}
+
+enum class NfcTagLocation {
+    DESK, REMOTE
+}
```

Comment:
I'm not sure if this is the best location for this enum, but, it's better to use an enum instead of `String` because it constrains the allowable values, is easier to figure out what those values should be, and generally faster and more data compact